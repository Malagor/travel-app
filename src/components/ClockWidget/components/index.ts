export { Clock } from './Clock'
export { DigitalWatch } from './DigitalWatch'
export { DateString  } from './Date'