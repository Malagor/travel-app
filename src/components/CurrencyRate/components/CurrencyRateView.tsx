import React, { FC } from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import { CURRENCY_MAP } from 'appConstants/currencyMap';

type CurrencyRateViewProps = {
  currentCountry: string;
  rates: [string, number][];
};

function createData(rate: number, currencyCode: string) {
  return { rate, currencyCode };
}

function createRows(ratesList: [string, number][]) {
  return ratesList.map((rate) => createData(rate[1], rate[0]));
}

export const CurrencyRateView: FC<CurrencyRateViewProps> = ({
  currentCountry,
  rates,
}) => {
  const rows = createRows(rates);

  return (
    <Table aria-label="currency rate table">
      <TableHead>
        <TableRow>
          <TableCell align="center" colSpan={2}>
            {`1 ${CURRENCY_MAP[currentCountry]} =`}
          </TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {rows.map((row) => (
          <TableRow key={row.currencyCode}>
            <TableCell component="th" scope="row" align="right">
              {row.rate}
            </TableCell>
            <TableCell>{row.currencyCode}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};
