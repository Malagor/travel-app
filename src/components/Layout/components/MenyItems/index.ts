export { mainListItems, secondaryListItems } from './MenuItems';
