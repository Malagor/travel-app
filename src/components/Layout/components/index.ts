export { Footer } from './Footer'
export { Header } from './Header'
export { SideBar } from './SideBar'
