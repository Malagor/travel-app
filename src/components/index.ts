export { App } from './App';
export { Clock } from './Clock';
export { ErrorMessage } from './ErrorMessage'
export { PointInfoTest } from './PointInfoTest';
