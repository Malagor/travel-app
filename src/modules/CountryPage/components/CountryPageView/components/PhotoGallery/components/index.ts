export { NextArrow } from './NextArrow';
export { PrevArrow } from './PrevArrow';
export { Slide } from './Slide';
