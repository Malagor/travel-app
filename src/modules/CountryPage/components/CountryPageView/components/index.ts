export { PhotoGallery } from './PhotoGallery';
export { VideoPlayer } from './VideoPlayer';
export { Map } from './Map';
