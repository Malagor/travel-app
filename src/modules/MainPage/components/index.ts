export { PageTitle } from './PageTitle/PageTitle';
export { CountryList } from './CountryList';
