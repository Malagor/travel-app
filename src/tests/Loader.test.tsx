import React from 'react';
import { shallow } from 'enzyme';
import { Loader } from 'components';

it('expect to render the Loader component', () => {
  expect(shallow(<Loader />)).toMatchSnapshot();
});
