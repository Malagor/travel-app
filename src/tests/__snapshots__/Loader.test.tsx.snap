// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render the Loader component 1`] = `
<div
  className="loader"
>
  <div
    className="cssload-loader"
  >
    <div
      className="cssload-inner cssload-one"
    />
    <div
      className="cssload-inner cssload-two"
    />
    <div
      className="cssload-inner cssload-three"
    />
  </div>
</div>
`;
